textbox anal

wak3upalex opened the pull request:
# GOAL

wak3upalex committed:
changed comment

wak3upalex committed:
start_date selection

wak3upalex committed:
extraction of data in rectangles/textboxes is now in function + `company_name` and `start_date` were changed to get data from function

wak3upalex committed:
import of necessary information from textboxes

Fascinat0r committed:
Create sonar-project.properties

Fascinat0r committed:
Create build.yml

Fascinat0r committed:
Merge pull request #2 from EstateMonitor/Add-CI

Add ci

Fascinat0r commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,27 @@
+      - uses: sonarsource/sonarqube-scan-action@master
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
```
Крутой CI, кто его написал? Он молодец

wak3upalex replied:
Вижу впервые, поэтому будь готов отвечать на возможные вопросы. Могут быть глупыми, но глупых вопросов не существует

Fascinat0r commented on main.py:
```diff
@@ -1,5 +1,13 @@
 import fitz  # импортируем библиотеку pymupdf
 
+
+def select_extract_data(x0, y0, x1, y1, r, g, b, page):
```
Функция выполняет две задачи - извлечение текста и разметка области.
Ты же можешь сразу передавать rect....

Fascinat0r replied:
А рисовать отдельно, тоже передавая rect

wak3upalex replied:
Исправил в обновленном коде

Fascinat0r commented on main.py:
```diff
@@ -1,5 +1,13 @@
+
 def highlight_sorted_drawings(pdf_path, output_path):
     # Открываем документ
     doc = fitz.open(pdf_path)
```
Нужно добавить проверку на существование файла и обработку ошибок при его открытии

wak3upalex replied:
А из ошибок что может быть? Файл пустой, файл не того типа?

wak3upalex replied:
Добавил первичный вид обработки ошибок. Подтяну логгирование и будет адекватно сохраняться

Fascinat0r commented on main.py:
```diff
@@ -13,7 +21,7 @@ def highlight_sorted_drawings(pdf_path, output_path):
             height = round(rect.height, 1)
-            if height == 3:  # Проверка, чтобы высота рисунка была больше 3
+            if height == 3:  # Проверка, чтобы высота рисунка была равна 3
                 if height not in drawings_dict:
```
Ну пиздец тут уровней вложенности, невозможно читать

Fascinat0r commented on main.py:
```diff
@@ -34,16 +42,38 @@ def highlight_sorted_drawings(pdf_path, output_path):
             print(f"Page {page_num + 1}: Rectangle {rect} with height {height} highlighted in color {color}")
 
-            # organization_name of OOO
-            top_rect = fitz.Rect(rect.x0 + 90, rect.y0 - 15, rect.x1 - 130, rect.y1)
```
Вынеси взятие компании в отдельную функцию, а константы переделай в rect. Их же можно просто складывать и получать результирующие координаты

Fascinat0r commented on main.py:
```diff
@@ -34,16 +42,38 @@ def highlight_sorted_drawings(pdf_path, output_path):
-            page.draw_rect(top_rect, color=(0, 1, 0), fill_opacity=0.1)  # Размечаем область выше рисунка
-            print(f"Extracted organization_name above rectangle: {organization_name}")
+            # company_name of OOO
+            company_name = select_extract_data(rect.x0 + 90, rect.y0 - 15, rect.x1 - 130, rect.y1, 0,1,0, page)
```
Создай словарь, в котором на каждое название поля будут приходится эти координаты, лучше в виде rect сразу.
И тогда ты сможешь сделать из этого цикл по ключам

Fascinat0r replied:
Уточню, сделать цикл это тоже претензия

Fascinat0r commented on main.py:
```diff
@@ -34,16 +42,38 @@ def highlight_sorted_drawings(pdf_path, output_path):
+
 
     # Сохраняем документ с изменениями
     doc.save(output_path)
```
Добавь обработку ошибок при сохранении документа.

Fascinat0r replied:
def save_document(doc, output_path):
    try:
        doc.save(output_path)
        logging.info(f"Modified PDF saved as: {output_path}")
    except Exception as e:
        logging.error(f"Error saving PDF: {e}")
        raise

Fascinat0r commented on main.py:
```diff
@@ -34,16 +42,38 @@ def highlight_sorted_drawings(pdf_path, output_path):
 
+
 # Укажите путь к вашему PDF файлу и путь для сохранения измененного PDF
 pdf_path = "1.pdf"
```
Вынеси это всё к хуям в отдельную функцию, это явно должны быть параметры к этой функции, по обработке одного файла

Fascinat0r commented on main.py:
```diff
@@ -13,7 +21,7 @@ def highlight_sorted_drawings(pdf_path, output_path):
         for drawing in drawings:
```
строка 16, range(len(doc))

почему не enumerate(doc)?

Fascinat0r replied:
И ещё ты кучу раз делаешь load page, это пиздец.
Инициируй их сразу
pages = [doc.load_page(page_num) for page_num in range(len(doc))] например

wak3upalex committed:
coordinates of rectangles for highlighting are got via `rect` instead of one by one

wak3upalex committed:
simplified work with colours

wak3upalex committed:
added a separate function `highlight_rect` to draw rectangles around data for debugging

wak3upalex committed:
minor refactoring

wak3upalex committed:
errors handling demo prototype (will be fixed with added `logging`)

Fascinat0r committed:
Add offsets

wak3upalex committed:
fixed minor coordinate position

wak3upalex committed:
Wrap work with pdf in a class

wak3upalex committed:
Initialize logging

wak3upalex committed:
Add logging to `validate_pdf`, `highlight_rect`, `select_extract_data`

wak3upalex committed:
Simplify function `highlight_sorted_drawings`

For simplicity, I decided to separate code, which extracts rectangles form the file and group them in one dictionary. New function `extract_and_group_rects` do this functionality.

Now Code Complexity of the highlight sorted drawings function is 75% (used to be 212%)

wak3upalex committed:
Simplify function `highlight_sorted_drawings`

For simplicity, I decided to separate code on two more different functions:
+ `highlight_drawings`
  + This function allows to paint necessary rectangles with data to visualize fields program working with
+ `extract_field_data`
  + This function indicate necessary fields to scan from specialized PDF file with data and extract data from these fields

Now Code Complexity of the `highlight_sorted_drawings` function is 0% (used to be 212%)

Fascinat0r committed:
Draft

Fascinat0r committed:
Fix config file

Fascinat0r committed:
Add requirements.txt

Fascinat0r committed:
Add some exceptions

Fascinat0r committed:
Fix pages to page_nums

Fascinat0r committed:
Specify the type of fields in Rect model

Fascinat0r committed:
Delete unnecessary cmd file

Fascinat0r committed:
Rebuild table handler

Fascinat0r committed:
Add draw ability

Fascinat0r committed:
Fix config file, it works

Fascinat0r committed:
Implement draw ability

Fascinat0r committed:
Optimize table handler

Fascinat0r committed:
Convert reports dict to DTO

Fascinat0r committed:
Add multi-page table support

Fascinat0r committed:
Cleanup service layer

Fascinat0r committed:
Add A LOT of comments
